File: semgrep-core/tests/go/dots_method_chaining.go
func foo() {
    
  //ERROR: match
  f = o.foo().m().h().bar().z()

  //ERROR: match
  f = o.foo().bar()

  // this one does not contain the bar()
  f = o.foo().m().h().z()

  //ERROR: match $O can match o.before()
  f = o.before().foo().m().h().bar().z()

}

File: semgrep-core/tests/scala/dots_method_chaining.scala
    
//ERROR: match
def f = o.foo().m().h().bar().z()

//ERROR: match
def f = o.foo().bar()

  // this one does not contain the bar()
def f = o.foo().m().h().z()

  //ERROR: match $O can match o.before()
def f = o.before().foo().m().h().bar().z()

